admin-ui/plugins/fido/redux/sagas/FidoSaga.ts

```diff
@@ -0,0 +1,122 @@
+  const issuer: string = yield select((state: RootState) => state.authReducer.issuer)
+  const configApi = new JansConfigApi.Fido2ConfigurationApi(getClient(JansConfigApi, token, issuer))
+  const registrationApi = new JansConfigApi.Fido2RegistrationApi(
+    getClient(JansConfigApi, token, issuer),
+  )
+  return new FidoApi(configApi, registrationApi)
+}
+
+function* errorToast({ error }: ErrorToastAction): SagaIterator<void> {
+  yield put(
+    updateToast(
+      true,
+      'error',
+      error?.response?.data?.description || error?.response?.data?.message || error.message,
+    ),
+  )
+}
+
+export function* updateFidoSaga({
+  payload,
+}: UpdateFidoAction): SagaIterator<AppConfiguration | ApiError> {
+  const audit: AuditLog = yield call(initAudit)
+  try {
+    const fidoApi: FidoApi = yield call(createFidoApi)
+    const data: AppConfiguration = yield call(fidoApi.putPropertiesFido2, {
+      appConfiguration: payload,
+    })
+    yield put(updateToast(true, 'success'))
+    yield put(getFidoConfiguration())
+    yield call(postUserAction, audit)
+    return data
+  } catch (e) {
+    const error = e as ApiError
+    yield put(updateToast(true, 'error'))
+    if (isFourZeroOneError(error)) {
+      const jwt: string | null = yield select((state: RootState) => state.authReducer.userinfo_jwt)
+      yield put(getAPIAccessToken(jwt))
+    }
+    return error
+  }
+}
+
+export function* getFidoSaga(): SagaIterator<AppConfiguration | ApiError> {
+  const audit: AuditLog = yield call(initAudit)
+  try {
+    const fidoApi: FidoApi = yield call(createFidoApi)
+    const data: AppConfiguration = yield call(fidoApi.getPropertiesFido2)
+    yield put(getFidoConfigurationResponse(data))
+    yield call(postUserAction, audit)
+    return data
+  } catch (e) {
+    const error = e as ApiError
+    yield put(getFidoConfigurationResponse({} as AppConfiguration))
+    if (isFourZeroOneError(error)) {
+      const jwt: string | null = yield select((state: RootState) => state.authReducer.userinfo_jwt)
+      yield put(getAPIAccessToken(jwt))
+    }
+    return error
+  }
+}
+
+export function* deleteFido2DeviceData({
+  payload,
+}: DeleteFido2DeviceAction): SagaIterator<void | ApiError> {
+  const audit: AuditLog = yield call(initAudit)
+  try {
+    const fidoApi: FidoApi = yield call(createFidoApi)
+    const data: void = yield call(fidoApi.deleteFido2DeviceData, payload)
+    yield put(updateToast(true, 'success'))
+    yield put(deleteFido2DeviceDataResponse())
+    yield call(postUserAction, audit)
+    return data
+  } catch (error) {
+    yield call(errorToast, { error: error as ApiError })
+    yield put(deleteFido2DeviceDataResponse())
+    return error as ApiError
+  }
+}
+
+export function* watchGetFido(): SagaIterator<void> {
+  yield takeEvery('fido2/getFidoConfiguration', getFidoSaga)
+}
+
+export function* watchUpdateFido(): SagaIterator<void> {
+  yield takeLatest('fido2/putFidoConfiguration', updateFidoSaga)
+}
```

Comment:
Prefer takeLatest for read to avoid stale overwrites

Multiple “get” triggers can return out of order. takeLatest guards against stale responses clobbering fresh state.

Apply:

```diff
 export function* watchGetFido(): SagaIterator<void> {
-  yield takeEvery('fido2/getFidoConfiguration', getFidoSaga)
+  yield takeLatest('fido2/getFidoConfiguration', getFidoSaga)
 }
```